internal/config/merge.go

```diff
@@ -0,0 +1,36 @@
+// Copyright 2024 Adevinta
+
+package config
+
+import (
+	"fmt"
+
+	"dario.cat/mergo"
+)
+
+// Merger has the ability to merge two configurations.
+type Merger interface {
+	Merge(dst, src Config) (Config, error)
+}
+
+// LavaMerger represents a merger for the Lava configuration.
+type LavaMerger struct{}
+
+// Merge merges two configurations. The values of the configuration
+// passed as first parameter will override those in the configuration
+// passed as second parameter.
+func (lm LavaMerger) Merge(dst, src Config) (Config, error) {
```

Comment:
Why do you need the interface? We don't plan to implement more mergers, right?

We can even make the function private if it is not used outside the `config` package. Also remember to update the doc comment to reflect the new behavior.


```suggestion
// Merge merges two configurations. The values of the configuration
// passed as src will override those in dst.
func Merge(dst, src Config) (Config, error) {
```